doc/configuration/authentication_modes.rst

```diff
@@ -0,0 +1,133 @@
+  first step, the plugin triggers a challenge. privacyIDEA sends the challenge
+  response --- a fresh OTP value --- to the user via E-Mail.
+  In a second step, the user responds to the challenge by entering the
+  respective OTP value in the plugin's login form. The plugin sends the
+  challenge response to privacyIDEA, which decides whether the authentication
+  is valid or not.
+* The PUSH token type implements the ``outofband`` mode. With a PUSH token,
+  the authentication step also consists of two steps:
+  In a first step, the user triggers a challenge. privacyIDEA pushes the
+  challenge to the user's smartphone app. In a second step, the user approves
+  the challenge on their phone, and the app responds to the challenge on behalf
+  of the user. The plugin periodically queries privacyIDEA to check if
+  the challenge has been answered correctly and the authentication is valid.
+
+The following describes the authentication flows of the three authentication
+modes in more detail.
+
+.. _authentication_mode_authenticate:
+
+``authenticate`` mode
+---------------------
+
+ * The user enters a OTP PIN along with an OTP value.
+ * The plugin sends a request to the ``/validate/check`` endpoint:
+
+   .. code-block:: text
+
+     POST /validate/check
+
+     user=<user>&pass=<PIN+OTP>
+
+  and privacyIDEA returns whether the authentication request has succeeded
+  or not.
+
+.. _authentication_mode_challenge:
+
+``challenge`` mode
+------------------
+
+ * The plugin triggers a challenge, for example via the
+   ``/validate/triggerchallenge`` endpoint:
+
+   .. code-block:: text
+
+     POST /validate/triggerchallenge
+
+     user=<user>
+
+   The plugin receives a transaction ID which we call ``transaction_id`` and
+   asks the user for the challenge response.
+ * The user enters the challenge response, which we call ``OTP``.
+   The plugin forwards the response to privacyIDEA along with the
+   transaction ID:
+
+   .. code-block:: text
+
+     POST /validate/check
+
+     user=<user>&transaction_id=<transaction_id>&pass=<OTP>
+
+  and privacyIDEA returns whether the authentication request succeeded or not.
+
+.. _authentication_mode_outofband:
+
+``outofband`` mode
+------------------
+
+ * The plugin triggers a challenge, for example via the
+   ``/validate/triggerchallenge`` endpoint:
+
+   .. code-block:: text
+
+     POST /validate/triggerchallenge
+
+     user=<user>
+
+   The plugin receives a transaction ID which we call ``transaction_id``.
+   The plugin may now periodically query the status of the challenge by
+   polling the ``/validate/polltransaction`` endpoint:
+
+   .. code-block:: text
+
+     GET /validate/polltransaction
+
+     transaction_id=<transaction_id>
+
+   If this endpoint returns ``false``, the challenge has not been answered yet.
+ * The user approves the challenge on a separate device, e.g. their
+   smartphone app. The app communicates with a tokentype-specific endpoint
```

Comment:
tokentype-specific endpoint of privacyIDEA!

Reply:
fixed in 076624be5af5d799ff9e8cdf6e6d362f1781e495